gpytorch/kernels/kernel.py

```diff
@@ -187,42 +187,28 @@ def forward(self, x1, x2, diag=False, batch_dims=None, **params):
         """
         raise NotImplementedError()
 
-    def _create_input_grid(self, x1, x2, diag=False, batch_dims=None, **params):
-        """
-        This is a helper method for creating a grid of the kernel's inputs.
-        Use this helper rather than maually creating a meshgrid.
-
-        The grid dimensions depend on the kernel's evaluation mode.
+    def _covar_sq_dist(self, x1, x2, **params):
+        if params.get('batch_dims') == (0, 2):
+            x1 = x1.unsqueeze(0).transpose(0, -1)
+            x2 = x2.unsqueeze(0).transpose(0, -1)
 
-        Args:
```

Comment:
It might be good to whiten x1 and x2 beforehand? This might improve the stability.